client/extensions/woocommerce/app/promotions/index.js

```diff
@@ -23,23 +23,23 @@ class Promotions extends Component {
 	static propTypes = {
 		site: PropTypes.shape( {
 			ID: PropTypes.number,
-		} )
+		} ),
 	};
 
 	componentDidMount() {
 		const { site } = this.props;
 		if ( site && site.ID ) {
-			this.props.fetchCoupons( site.ID, { page: 1 } );
+			this.props.fetchPromotions( site.ID );
 		}
 	}
 
 	componentWillReceiveProps( newProps ) {
 		const { site } = this.props;
 		const newSiteId = newProps.site && newProps.site.ID || null;
 		const oldSiteId = site && site.ID || null;
+
 		if ( oldSiteId !== newSiteId ) {
-			// TODO: Fill in with current page number.
-			this.props.fetchCoupons( newSiteId, { page: 1 } );
+			this.props.fetchPromotions( site.ID );
```

Comment:
Did you mean to pass `newSiteId` instead?

Reply:
good catch!

Reply:
Thanks!